src/WebJobs.Script/StorageProvider/HostAzureTableStorageProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using Azure.Core.Pipeline;
+using Azure.Data.Tables;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    internal sealed class HostAzureTableStorageProvider : IAzureTableStorageProvider
+    {
+        private readonly ILogger<HostAzureTableStorageProvider> _logger;
+        private readonly IDelegatingHandlerProvider _delegatingHandlerProvider;
+        private readonly TableServiceClientProvider _tableServiceClientProvider;
+        private readonly IConfiguration _configuration;
+
+        public HostAzureTableStorageProvider(IConfiguration configuration, ILogger<HostAzureTableStorageProvider> logger, AzureComponentFactory componentFactory, AzureEventSourceLogForwarder logForwarder, IDelegatingHandlerProvider delegatingHandlerProvider)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _delegatingHandlerProvider = delegatingHandlerProvider;
+            _tableServiceClientProvider = new TableServiceClientProvider(componentFactory, logForwarder);
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        }
+
+        public bool TryCreateTableServiceClient(string connection, out TableServiceClient client)
+        {
+            if (string.IsNullOrEmpty(connection))
+            {
+                throw new ArgumentException($"'{nameof(connection)}' cannot be null or empty.", nameof(connection));
+            }
+
+            try
+            {
+                DelegatingHandler handler = _delegatingHandlerProvider?.Create();
```

Comment:
Can/should _delegatingHandlerProvider ever be null?l Looks like it's a DI injected ctor param

Reply:
With our current usage, it cannot be necessary since, as you mentioned, it's a DI injected constructor parameter. I didn't enforce the null check on the constructor because it's not essential for the provider to have a delegating handler provider—opting out could imply that one isn't needed. On second thought based on how we're using it, I'll enforce the null check on the ctor.